src/coord/src/coord.rs

```diff
@@ -942,12 +943,14 @@ impl Coordinator {
                             .first()
                             .expect("known to exist");
 
-                        self.catalog
-                            .compact_timestamp_bindings(source_id, compaction_ts)
-                            .expect("compacting timestamp bindings cannot fail");
+                        timestamp_compactions.push((source_id, compaction_ts));
                     }
                 }
 
+                self.catalog
+                    .compact_timestamp_bindings(&timestamp_compactions)
+                    .expect("compacting timestamp bindings cannot fail");
```

Comment:
if non-empty? or is it cheap to do a no-op transaction?

Reply:
Ah yeah, should be super cheap. The `if` statement is probably a good idea, but I'm just going to merge as is because CI is green.